tests/lib/rules/camelcase.js

```diff
@@ -79,6 +79,20 @@ ruleTester.run("camelcase", rule, {
         {
             code: "import { no_camelcased as camelCased, anoterCamelCased } from \"external-module\";",
             parserOptions: { ecmaVersion: 6, sourceType: "module" }
+        },
+        {
+            code: "const { no_camelcased = false } = bar;",
+            options: [{ properties: "never" }],
```

Comment:
I don't think we need the `properties` any more in most of these valid tests. Maybe one with `properties: "always"` and one with `properties: "never"` at the end of the test suite to ensure they don't affect the outcome?